mne/event.py

```diff
@@ -1422,3 +1424,82 @@ def get_condition(self, raw, condition=None, stim_channel=None, mask=None,
             conds_data.append(dict(events=cat_t0, event_id=cat_id,
                                    tmin=tmin, tmax=tmax))
         return conds_data[0] if len(conds_data) == 1 else conds_data
+
+
+def match_event_names(event_names, keys, *, on_missing='raise'):
+    """Search a collection of event names for matching (sub-)groups of events.
+
+    This function is particularly helpful when using grouped event names
+    (i.e., event names containing forward slashes ``/``). Assuming for example
+    the following grouped event names in the data, you could easily query for
+    all ``auditory`` and ``left`` event names::
+
+        >>> event_names = [
+        ...     'auditory/left',
+        ...     'auditory/right',
+        ...     'visual/left',
+        ...     'visual/right'
+        ... ]
+        >>> match_event_names(
+        ...     event_names=event_names,
+        ...     keys=['auditory', 'left']
+        ... )
+        ['auditory/left', 'auditory/right', 'visual/left']
```

Comment:
Long function descriptions like this should be discouraged -- ideally not more than "a few sentences" (and even then avoided if possible I think):

https://numpydoc.readthedocs.io/en/latest/format.html#extended-summary

To me what you have here should go in `Notes` or `Examples`. Probably the latter since most of the contents follows a "Assuming for example"

Reply:
I know that this is discouraged, but I didn't feel that I could explain what the function does without providing a working example. Therefore, I included said example at the top of the docstring.

Reply:
… meaning I'd like not to change it if you can live with that.

Reply:
@larsoner suggests to move to example section not to delete it. I agree it’s more standard

Reply:
See example https://github.com/mne-tools/mne-python/blob/47051833f21bb372d60afc3adbf4305648ac7f69/mne/event.py#L621

Reply:
I did understand that and know it's more standard, I just doubt it will help the users who will have to scroll down to the examples section to figure out what this function is even good for. Therefore I'm -1 on blindly following a standard just for the sake of it while forgetting about our users

Reply:
@agramfort The example you linked to is super long, which is not the case for this docstring here

Reply:
> on blindly following a standard just for the sake of it while forgetting about our users

I'm not pushing for this just for the sake of following the standard -- there are tradeoffs either way we go. By making the description really long you make some things easier (e.g., seeing an example) but other things harder (e.g., finding/reading about the parameters and return values). Moreover, we don't know all ways users will come to see and use a function -- we usually try to expose people to functionality through thorough and clear tutorials, which makes providing examples *in docstrings themselves* much less important. To me it's still better to have this in a dedicated Examples section.